src/assignments/scrabble-scorer.rst

```diff
@@ -0,0 +1,197 @@
+to work on Mars rover code already?
+
+The current Scrabble Scorer, is a program that will take in a word and return the point value.
+The program needs to be rewritten for two reasons. First the data structure used to store
+the letter point values is inefficient. Second the program should have three scoring
+algorithms and allow the user to pick which algorithm to use. These new features will make
+the program more efficient and user friendly.
+
+
+Requirements
+------------
+
+#. Create an ``oldScoreKey`` object that contains the previous scoring system.
+#. Create a ``transform`` function which will return a ``newScoreKey`` object.
+#. Create ``scoringAlgorithms`` array to hold three scoring objects, which include the scoring functions.
+#. Create a ``initialPrompt`` function that asks the user which scoring algorithm to use.
+#. Create a ``runProgram`` function to contain the starting point for your running program.
+
+Example Output
+^^^^^^^^^^^^^^
+::
+
+   Welcome to the Scrabble score calculator. Use ctrl+c to quit.
+
+   Which scoring algorithm would you like to use?
+
+   0 - Scrabble: The traditional scoring algorithm.
+   1 - Simple Score: Each letter is worth 1 point.
+   2 - Bonus Vowels: Vowels are 3 pts, consonants are 1pt
+   Enter 0,1,2: 0
+   Using algorithm: Scrabble
+
+   Enter a word to be scored:  LaunchCode
+   Score for 'LaunchCode': 18
+
+   Enter a word to be scored:  Rocket
+   Score for 'Rocket': 12
+
+   Enter a word to be scored:
+
+
+Detailed Requirements
+---------------------
+
+Transform
+^^^^^^^^^
+Currently the Scrabble Grader software uses the below data structure for the traditional
+Scrabble scoring algorithm. Take a few moments to review how ``oldScoreKey`` relates a
+letter to a point value.
+
+.. sourcecode:: js
+   :linenos:
+
+   const oldScoreKey = {
+      1: ['A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T'],
+      2: ['D', 'G'],
+      3: ['B', 'C', 'M', 'P'],
+      4: ['F', 'H', 'V', 'W', 'Y'],
+      5: ['K'],
+      8: ['J', 'X'],
+      10: ['Q', 'Z']
+   };
+
+The object keys of ``oldScoreKey`` are the Scrabble points and the values
+are arrays of letters. All letters in the array have the Scrabble
+point value equal to the key. Example ``'A'`` is worth 1 and ``'J'`` is worth 8.
+
+Write a ``transform`` function that takes in as a parameter the ``oldScoreKey``
+object. ``transform(oldScoreKey)`` will return an object with the letters as keys. The keys
+in the ``newScoreKey`` object should be *lowercase* letters.
+
+The ``newScoreKey`` object will have 26 keys, one for each letter. The value associated to each key
+will be the Scrabble point value.
+
+Example of new key storage
+
+* ``a`` is worth ``1``
+* ``j`` is worth ``8``
+* ``k`` is worth ``10``
+
+The new data structure for storing point values is more efficient. Can you think of why?
+With the old data structure, to find the point value for a letter you would need to iterate
```

Comment:
With the old format, to find the point value for a letter, the program must iterate over each key in ``oldScoreKey`` and check if the letter is inside the array paired with that key.  This "search within a search" is inefficient.

In the new data structure, the letters themselves are keys, so a single search will identify a point value.